src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -974,16 +974,28 @@ export function calculateLayout(visibleAttributes: ChartMeta[], cols: number, cu
     // sort the visibleAttributes by priority
     visibleAttributes.sort(chartMetaComparator);
 
+    const visibleAttributeSet = _.keyBy(visibleAttributes,attribute=>attribute.uniqueKey)
```

Comment:
What's the purpose of this?

Reply:
it's to check whether the charts in `currentGridLayout` are visible. if chart is not visible then discard it's layout. I'll add a comment